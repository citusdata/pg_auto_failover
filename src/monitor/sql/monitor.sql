-- Copyright (c) Microsoft Corporation. All rights reserved.
-- Licensed under the PostgreSQL License.

\x on

select *
  from pgautofailover.register_node('default', 'localhost', 9876, 'postgres');

select *
  from pgautofailover.register_node('default', 'localhost', 9877, 'postgres');

select *
  from pgautofailover.node_active('default', 'localhost', 9876,
                                  current_group_role => 'single');

select *
  from pgautofailover.register_node('default', 'localhost', 9877, 'postgres');

table pgautofailover.formation;

-- dump the pgautofailover.node table, omitting the timely columns
select formationid, nodeid, groupid, nodehost, nodeport,
       goalstate, reportedstate, reportedpgisrunning, reportedrepstate
  from pgautofailover.node;

select * from pgautofailover.get_primary('unknown formation');
select * from pgautofailover.get_primary(group_id => -10);
select * from pgautofailover.get_primary();

select * from pgautofailover.get_primary('default', 0);
select * from pgautofailover.get_other_nodes('localhost', 9876);

select pgautofailover.remove_node('localhost', 9876);

table pgautofailover.formation;

-- dump the pgautofailover.node table, omitting the timely columns
select formationid, nodeid, groupid, nodehost, nodeport,
       goalstate, reportedstate, reportedpgisrunning, reportedrepstate
  from pgautofailover.node;

select * from pgautofailover.node_active('default', 'localhost', 9877);

-- should fail as there's no primary at this point
select pgautofailover.perform_failover();
