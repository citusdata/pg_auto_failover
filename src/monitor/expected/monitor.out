-- Copyright (c) Microsoft Corporation. All rights reserved.
-- Licensed under the PostgreSQL License.
\x on
select * from pgautofailover.register_node('default', 'localhost', 9876, 'postgres');
-[ RECORD 1 ]---------------+-------
assigned_node_id            | 1
assigned_group_id           | 0
assigned_group_state        | single
assigned_candidate_priority | 100
assigned_replication_quorum | t

select * from pgautofailover.register_node('default', 'localhost', 9877, 'postgres');
ERROR:  ProceedGroupState couldn't find the primary node in formation "default", group 0
DETAIL:  activeNode is localhost:9877 in state wait_standby
select * from pgautofailover.node_active('default', 'localhost', 9876);
-[ RECORD 1 ]---------------+-------
assigned_node_id            | 1
assigned_group_id           | 0
assigned_group_state        | single
assigned_candidate_priority | 100
assigned_replication_quorum | t

select * from pgautofailover.node_active('default', 'localhost', 9877);
ERROR:  node localhost:9877 is not registered
table pgautofailover.formation;
-[ RECORD 1 ]--------+---------
formationid          | default
kind                 | pgsql
dbname               | postgres
opt_secondary        | t
number_sync_standbys | 1

-- dump the pgautofailover.node table, omitting the timely columns
select formationid, nodeid, groupid, nodename, nodeport,
       goalstate, reportedstate, reportedpgisrunning, reportedrepstate
  from pgautofailover.node;
-[ RECORD 1 ]-------+----------
formationid         | default
nodeid              | 1
groupid             | 0
nodename            | localhost
nodeport            | 9876
goalstate           | single
reportedstate       | init
reportedpgisrunning | t
reportedrepstate    | unknown

select * from pgautofailover.get_primary('unknown formation');
ERROR:  group has no writable node right now
select * from pgautofailover.get_primary(group_id => -10);
ERROR:  group has no writable node right now
select * from pgautofailover.get_primary();
ERROR:  group has no writable node right now
select * from pgautofailover.get_primary('default', 0);
ERROR:  group has no writable node right now
select * from pgautofailover.get_other_node('localhost', 9876);
ERROR:  function pgautofailover.get_other_node(unknown, integer) does not exist
LINE 1: select * from pgautofailover.get_other_node('localhost', 987...
                      ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select pgautofailover.remove_node('localhost', 9876);
-[ RECORD 1 ]--
remove_node | t

table pgautofailover.formation;
-[ RECORD 1 ]--------+---------
formationid          | default
kind                 | pgsql
dbname               | postgres
opt_secondary        | t
number_sync_standbys | 1

-- dump the pgautofailover.node table, omitting the timely columns
select formationid, nodeid, groupid, nodename, nodeport,
       goalstate, reportedstate, reportedpgisrunning, reportedrepstate
  from pgautofailover.node;
(0 rows)

select * from pgautofailover.node_active('default', 'localhost', 9876);
ERROR:  node localhost:9876 is not registered
-- should fail as there's no primary at this point
select pgautofailover.perform_failover();
ERROR:  cannot fail over: group does not have 2 nodes
